examples/python/ml/jax_kmeans.py

```diff
@@ -0,0 +1,176 @@
+    # k: number of clusters
+    # return: cluster centers, labels
+
+    data = jnp.append(data_alice, data_bob, axis=0)
+
+    n = data.shape[0]
+    m = data.shape[1]
+
+    # initialize cluster centers
+    centers = jnp.zeros((k, m))
+
+    def loop_for_1(i, stat):
+        centers = stat
+        centers = centers.at[i].set(data[random_numbers[i]])
+        return centers
+
+    centers = jax.lax.fori_loop(0, k, loop_for_1, centers)
+
+    # initialize labels
+    labels = jnp.zeros(n)
+
+    def loop_while_func(stat):
+
+        rounds, centers, new_centers, labels = stat
+        centers = new_centers
+
+        def loop_for_1(i, stat):
+            labels = stat
+            min_dist = jnp.linalg.norm(data[i] - centers[0])
+            labels = labels.at[i].set(0)
+
+            def loop_for_2(j, stat):
+                min_dist, labels = stat
+                dist = jnp.linalg.norm(data[i] - centers[j])
+                labels = labels.at[i].set(jnp.where(min_dist > dist, j, labels[i]))
+                min_dist = jnp.minimum(dist, min_dist)
+                return min_dist, labels
+
+            stat = (min_dist, labels)
+            min_dist, labels = jax.lax.fori_loop(1, k, loop_for_2, stat)
+            return labels
+
+        labels = jax.lax.fori_loop(0, n, loop_for_1, labels)
+
+        def loop_for_3(i, stat):
+            new_centers = stat
+            sums = jnp.zeros(m)
+
+            def loop_for_4(j, sums):
+                sums = sums + jnp.where(labels[j] == i, data[j], jnp.zeros(m))
+                return sums
+
+            sums = jax.lax.fori_loop(0, n, loop_for_4, sums)
+            avg = sums / jnp.sum(jnp.where(labels == i, 1, 0))
+            new_centers = new_centers.at[i].set(avg)
+            return new_centers
+
+        new_centers = jax.lax.fori_loop(0, k, loop_for_3, jnp.zeros((k, m)))
+
+        return rounds + 1, centers, new_centers, labels
+
+    stat = (0, jnp.zeros((k, m)), centers, labels)
+    res = jax.lax.while_loop(lambda stat: stat[0] < n_epochs, loop_while_func, stat)
+
+    return res[1], res[3]
+
+
+@ppd.device("P1")
+def points_from_alice(npoints):
+    np.random.seed(0xDEADBEEF)
+    return np.random.normal(size=(npoints, 2)) * 5
+
+
+@ppd.device("P2")
+def points_from_bob(npoints):
+    np.random.seed(0xC0FFEE)
+    return np.random.normal(size=(npoints, 2)) * 5
+
+
+def run_on_cpu():
+    # Two party share a dataset, and use kmeans to cluster them.
+
+    n_points_each_party = 30
+    k = 2
+    n_epochs = 10
+
+    np.random.seed(0xDEADBEEF)
+    x = np.random.normal(size=(n_points_each_party, 2)) * 5
+
+    np.random.seed(0xC0FFEE)
+    y = np.random.normal(size=(n_points_each_party, 2)) * 5
+
+    np.random.seed(0xCAFEBABE)
+    random_numbers = np.random.randint(n_points_each_party * 2, size=(k,))
+
+    centers, labels = jax.jit(kmeans, static_argnums=(2, 4))(
+        x, y, k, random_numbers, n_epochs
+    )
+
+    print(centers)
+    print(labels)
+    print()
```

Comment:
This line seems to be useless? The same as line 169

Reply:
Yeah I just want to show the correctness of the output from spu version, compared to cpu version.

Reply:
Or you mean the `print()`？

Reply:
> Or you mean the `print()`？

Yeah, I think it's better to delete this empty print

Reply:
> 

yeah, thats great.